CONTRIBUTING.md

```diff
@@ -146,9 +146,9 @@ of binaries in ./build directory.
 
 In order to test locally, you'd need a test setup. Local test setup automation is planned but but not done yet, so currently the environment has to be set up manually.
 
-Test environment  typically consist of 1 ESX and 2  guest VMs running inside of the
-ESX. We also support 1 ESX and 1 guest VM. We require ESX 6.0 and later,
-and a Linux VM running  Docker 1.10+ enabled for  plain text TCP connection, i.e.
+Test environment  typically consist of 1 ESX and 3 guest VMs running inside of the
```

Comment:
Agree with Mark. Nirdesh and I discussed this and we agree it's too heavy to have the basic precheckin tests running on a swarm cluster (at least 3 nodes) environment. For internal developers, we can use nimbus to deploy a testbed with one command. But for external developers, it will be too complicated to set up the environment simply for running the precheckin test.

On the other hand, we also want to make sure our CI can capture most common issues by running all P0 and part of P1 test cases. Also we should still support running all complicated test cases locally with some optional parameters

To achieve these different goals, here's our plan:
1. Developers should be able to run test-all on a single VM. This is achieved by moving all extensive test cases (which require complicated setup) to one or more special labels
2. Developers should be able to run complicated test case by specifying some optional parameters. The assumption is that the environment should be ready for running those tests
3. CI should continue to all extensive test cases as we have already planned

Let me know if you have any comments @msterin @shuklanirdesh82 . I will file a issue to track this task.